### PR TITLE
[record-minmax] Introduce ListFileIterator

### PR DESCRIPTION
**Pull request by jinevening**
This introduces an iterator for list format.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/14255
Draft PR: https://github.com/Samsung/ONE/pull/14257

### CODE REVIEW

File: compiler/record-minmax/src/ListFileIterator.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+std::vector<DataBuffer> ListFileIterator::next()
+{
+  const auto line = _lines.at(_curr_idx++);
+
+  const auto file_names = parse_line(line);
+
+  std::vector<DataBuffer> res;
+
+  // Have multiple files in one line
+  if (file_names.size() == _input_nodes.size())
```
Comment (seanshpark) [inline compiler/record-minmax/src/ListFileIterator.cpp:100]: Just curious. if `_input_nodes.size() == 1`, comment looks confusing.

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/record-minmax/src/ListFileIterator.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/record-minmax/src/ListFileIterator.cpp:101]: I agree. I've updated the comment. PTAL

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**